src/gt4py/analysis/passes.py

```diff
@@ -260,7 +260,7 @@ def visit_Expr(self, node: gt_ir.Expr):
             return []
 
         def visit_VarRef(self, node: gt_ir.VarRef):
-            result = [(node.name, None)]
+            result = [(node.name, node.index)]
```

Comment:
The index is not used at all, so I think it's better to remove it and return None as before

Reply:
I suppose you're right - the index is `None` naturally for scalars, but if it is not `None`, the code calling into `visit_VarRef` does not use that. I can revert that section.